automated install/basic-install.sh

```diff
@@ -351,40 +351,51 @@ package_manager_detect() {
-        if ! grep -qiE 'fedora|fedberry' /etc/redhat-release; then
-            if rpm -qa | grep -qi 'epel'; then
-                printf "  %b EPEL repository already installed\\n" "${TICK}"
-            else
-                local RH_RELEASE EPEL_PKG
-                # EPEL not already installed, add it based on the release version
-                RH_RELEASE=$(grep -oP '(?<= )[0-9]+(?=\.?)' /etc/redhat-release)
-                EPEL_PKG="https://dl.fedoraproject.org/pub/epel/epel-release-latest-${RH_RELEASE}.noarch.rpm"
-                printf "  %b Enabling EPEL package repository (https://fedoraproject.org/wiki/EPEL)\\n" "${INFO}"
-                "${PKG_INSTALL[@]}" "${EPEL_PKG}"
-                printf "  %b Installed %s\\n" "${TICK}" "${EPEL_PKG}"
-            fi
-        fi
+        if [ "${PKG_MANAGER}" = "dnf" ] || [ "${PKG_MANAGER}" = "yum" ] ; then
+          # CentOS package manager returns 100 when there are packages to update so we need to || true to prevent the script from exiting.
+          PKG_COUNT="${PKG_MANAGER} check-update | egrep '(.i686|.x86|.noarch|.arm|.src)' | wc -l || true"
+          INSTALLER_DEPS=(git dialog iproute newt procps-ng which chkconfig ca-certificates)
+          PIHOLE_DEPS=(cronie curl findutils sudo unzip libidn2 psmisc libcap nmap-ncat jq)
+          PIHOLE_WEB_DEPS=(lighttpd lighttpd-fastcgi php-common php-cli php-pdo php-xml php-json php-intl)
 
-    # If neither apt-get or yum/dnf package managers were found
+        # If the host OS is centos (or a derivative), epel is required for lighttpd
+          if ! grep -qiE 'fedora|fedberry' /etc/redhat-release; then
+              if rpm -qa | grep -qi 'epel'; then
+                  printf "  %b EPEL repository already installed\\n" "${TICK}"
+              else
+                  local RH_RELEASE EPEL_PKG
+                  # EPEL not already installed, add it based on the release version
+                  RH_RELEASE=$(grep -oP '(?<= )[0-9]+(?=\.?)' /etc/redhat-release)
+                  EPEL_PKG="https://dl.fedoraproject.org/pub/epel/epel-release-latest-${RH_RELEASE}.noarch.rpm"
+                  printf "  %b Enabling EPEL package repository (https://fedoraproject.org/wiki/EPEL)\\n" "${INFO}"
+                  "${PKG_INSTALL[@]}" "${EPEL_PKG}"
+                  printf "  %b Installed %s\\n" "${TICK}" "${EPEL_PKG}"
+              fi
+          fi
+        else
+          # openSUSE packages
+          # openSUSE package manager returns number when there are packages to update so we need to || true to prevent the script from exiting.
+          PKG_COUNT="${PKG_MANAGER} lu | egrep '(.i686|.x86|.noarch|.arm|.src)' | wc -l || true"
+          INSTALLER_DEPS=(git dialog iproute newt procps which ca-certificates)
+          PIHOLE_DEPS=(cronie curl findutils sudo unzip libidn2 psmisc libcap-ng0 nmap ncat jq)
+          PIHOLE_WEB_DEPS=(lighttpd php8 php8-fastcgi php8-cli php8-pdo php8-intl php8-openssl php8-sqlite)
```

Comment:
```suggestion
          PIHOLE_WEB_DEPS=(lighttpd php php-fastcgi php-cli php-pdo php-intl php-openssl php-sqlite)
```

Reply:
Please do not hard-code a `php` version. E.g. `leap` on `x86_64` has `php8` while on `arm` `php7.4 is the default.